### PR TITLE
[WIP] refactor(loader): v1.0.0

### PR DESCRIPTION
**Pull request by d3viant0ne**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, did you update the README?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

- Fixes for Webpack 4.x
- Drops support for NodeJS 4.x
- Sets minimum peerDep to Webpack 3.x

BREAKING CHANGE: Sets `engines` to `"node": ">= 6.9.0 || >= 8.9.0"`
BREAKING CHANGE: Drops support for Webpack `v2.x`

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
<h3 align="center"> ⚠️ DO NOT MERGE ⚠️ </h3>

### `Noteable Changes`

### `Features`

#### `url` (`options.url`)

**file.css**
```css
.selector {
  background-url: url('path/to/file.png');
}
```
**module.js**
```js
import CSS__URL__0 from 'path/to/file.png';

export default `.selector { background-url: url(${CSS__URL__0}); }`
```

#### `import` (`options.import`)

**file.css**
```css
@import './path/to/import.css';

...css
```
**module.js**
```js
import CSS__IMPORT__0 from 'path/to/import.css';

export default `${css}`
```

#### `sourceMap` (`options.sourceMap`)

- enable/disable source maps

### `BREAKING CHANGES`

#### `root` (`options.root`)

- Removed

#### `alias` (`options.alias`) 

- Removed (needs triage)

#### `modules` (`options.modules|options.localIdentName|options.camelCase`)

- Removed (will be a separate plugin usable via `postcss-loader`)

#### `minimize` (`options.minimize`)

- **Removed**

#### `importLoaders` (`options.importLoaders`)

- Removed (not needed anymore `webpack >= v2.0.0`)

### `Issues`

> Coming soon...

### `TODO`

- port changes from `src/lib/plugins/url` to `postcss-icss-url` (PR #?)
   - add plugin option to filter urls `{Regex|Function}`
- port changes from `src/lib/plugins/import` to `postcss-icss-import` (PR #?)
   - add plugin option to filter urls `{Regex|Function}`
- update CSS runtime (`src/lib/runtime.js`)

### `Status`

|Name|Status|
|:-----:|:-----:|
|Default| ✅ |
|CSS URLs| ✅ |
|CSS Imports|✅ |
|CSS Runtime|❔|
|Options - url | ⚠️|
|Options - import| ⚠️|
|Options - sourceMap| ✅ |

### `Blocked by`

> Coming soon...

**Comment by ZeeCoder** [comment]
@michael-ciniawsky any way I can help with this? I believe this would include custom exports support.

File: package.json [hunk 1]
```diff
@@ -2,50 +2,82 @@
...
-    "loader-utils": "^1.0.2",
-    "lodash.camelcase": "^4.3.0",
-    "object-assign": "^4.1.1",
-    "postcss": "^5.0.6",
-    "postcss-modules-extract-imports": "^1.2.0",
-    "postcss-modules-local-by-default": "^1.2.0",
-    "postcss-modules-scope": "^1.1.0",
-    "postcss-modules-values": "^1.3.0",
+    "loader-utils": "^1.1.0",
+    "postcss": "^6.0.15",
```
Comment (mikeumus) [inline package.json]: This line would solve https://github.com/webpack-contrib/css-loader/issues/691 in my case at least.

**Comment by michael-ciniawsky** [comment]
Closing in favor of #748 (for now).

> ⚠️ Please still **don't deleted** the branch yet

**Pull request closed without merging**